src/idlcxx/src/backendCpp11Type.c

```diff
@@ -386,16 +410,16 @@ get_potential_nr_discr_values(const idl_union_t *union_node)
       switch(node->mask & IDL_INTEGER_MASK_IGNORE_SIGN)
       {
       case IDL_INT8:
-        nr_discr_values = 0xff;
+        nr_discr_values = UCHAR_MAX;
         break;
       case IDL_INT16:
-        nr_discr_values = 0xffff;
+        nr_discr_values = USHRT_MAX;
         break;
       case IDL_INT32:
-        nr_discr_values = 0xffffffffU;
+        nr_discr_values = UINT_MAX;
         break;
       case IDL_INT64:
-        nr_discr_values = 0xffffffffffffffffULL;
+        nr_discr_values = ULONG_MAX;
```

Comment:
It is much better to use UINTx_MAX where x is 8, 16, 32 or 64. The 64-bit case is almost certainly not as intended on a 32-bit machine.